zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/repo/NotebookRepoSync.java

```diff
@@ -50,28 +52,60 @@
    * @param (conf)
    * @throws - Exception
    */
-  public NotebookRepoSync(ZeppelinConfiguration conf) throws Exception {
+  @SuppressWarnings("static-access")
+  public NotebookRepoSync(ZeppelinConfiguration conf) {
     config = conf;
 
     String allStorageClassNames = conf.getString(ConfVars.ZEPPELIN_NOTEBOOK_STORAGE).trim();
     if (allStorageClassNames.isEmpty()) {
-      throw new IOException("Empty ZEPPELIN_NOTEBOOK_STORAGE conf parameter");
+      allStorageClassNames = "org.apache.zeppelin.notebook.repo.VFSNotebookRepo";
+      LOG.warn("Empty ZEPPELIN_NOTEBOOK_STORAGE conf parameter, using default {}", defaultStorage);
     }
     String[] storageClassNames = allStorageClassNames.split(",");
     if (storageClassNames.length > getMaxRepoNum()) {
-      throw new IOException("Unsupported number of storage classes (" +
-        storageClassNames.length + ") in ZEPPELIN_NOTEBOOK_STORAGE");
+      LOG.warn("Unsupported number {} of storage classes in ZEPPELIN_NOTEBOOK_STORAGE : {}\n" +
+        "first {} will be used", storageClassNames.length, allStorageClassNames, getMaxRepoNum());
     }
 
-    for (int i = 0; i < storageClassNames.length; i++) {
+    for (int i = 0; i < Math.min(storageClassNames.length, getMaxRepoNum()); i++) {
       @SuppressWarnings("static-access")
-      Class<?> notebookStorageClass = getClass().forName(storageClassNames[i].trim());
+      Class<?> notebookStorageClass;
+      try {
+        notebookStorageClass = getClass().forName(storageClassNames[i].trim());
+        Constructor<?> constructor = notebookStorageClass.getConstructor(
+                  ZeppelinConfiguration.class);
+        repos.add((NotebookRepo) constructor.newInstance(conf));
+      } catch (ClassNotFoundException | NoSuchMethodException | SecurityException |
+          InstantiationException | IllegalAccessException | IllegalArgumentException |
+          InvocationTargetException e) {
+        LOG.warn("Failed to initialize {} notebook storage class {}", storageClassNames[i], e);
+      }
+    }
+    // couldn't initialize any storage, use default
+    if (getRepoCount() == 0) {
+      initializeDefaultStorage(conf);
```

Comment:
nit: Personally I prefer to leave log when fail-back occurs. I'm curious others also prefer it. I don't have string opinion about this, so it shouldn't block this PR.